### DISCUSSION TITLE
two side Windows for filenames like in Total Commander

### DISCUSSION THREAD

**Discussion by masterwishx** [WinMerge/winmerge#868] [Ideas]
Can we get 2 side windows for filenames like in Total Commander compare and Beyond Copmare ?
its very hard to understand differents when all files shown on one side

**Comment by sdottaka** [comment]
Unfortunately, WinMerge doesn't have that feature.

When Bitbucket previously had a WinMerge Mercurial repository, a pull request for that feature was posted.

However, the pull request only supported two-side mode, so I asked to support the original mode as well.

Unfortunately, the next fix wasn't posted and the bitbucket repository was gone.

**Reply by masterwishx** [reply]
So it's  unpossible to do?

**Reply by sdottaka** [reply]
It probably won't be implemented unless someone creates a pull request that supports both the two side mode and the original mode.

**Comment by GaziYucel** [comment]
Hi, this is a feature I would love too. I was thinking to implement this feature and post a pull request. Have however no experience with developing in WinMerge. Can you help me or redirect me to resources where I can make a quick start. Thanks.

**Reply by sdottaka** [reply]
Great!
Here are the source files for the folder comparison window

https://github.com/WinMerge/winmerge/blob/master/Src/DirView.cpp
https://github.com/WinMerge/winmerge/blob/master/Src/DirDoc.cpp
https://github.com/WinMerge/winmerge/blob/master/Src/DirFrame.cpp

As a reminder, WinMerge allows two folder comparisons and three folder comparisons, which means that up to three panes will be displayed. Since the original one-pane display is also useful, it would be helpful if you could implement a way to switch to the original one-pane display.

**Reply by GaziYucel** [reply]
I had the possibility of switching already in mind, because of your comment of July 2021. 

I was thinking about to make this choice through the settings or by a 'layout' option in the view menu. But need to investigate this, because I am not sure yet if the interface/app needs to be reloaded/restarted for this switch. 

Thank you for your quick response.

**Comment by spacecabbie** [comment]
is this ever going to be implemented ?

**Comment by ArthurQiangLi** [comment]
I would love to see this function (side-by-side view for two folders),  but 'a folder is also a file'. I was actually surprised that it only has the tree view for folders. I think it's a natural way to compare folders on the left and right sides.
